apps/rush-lib/src/cli/utilities/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -115,7 +115,21 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
       return undefined;
     }
 
-    return packageDescription.dependencies[dependencyName];
+    if (!packageDescription.dependencies.hasOwnProperty(dependencyName)) {
+      return undefined;
+    }
+
+    const version: string = packageDescription.dependencies[dependencyName];
+    // version will be either:
+    // A - the version (e.g. "0.0.5")
+    // B - a peer dep version (e.g. "/gulp-karma/0.0.5/karma@0.13.22"
+    //                           or "/sinon-chai/2.8.0/chai@3.5.0+sinon@1.17.7")
+
+    // check to see if this is the special style of specifiers
```

Comment:
>ch [](start = 7, length = 2)

Validate that there are exactly three slashes.

If it's an unexpected pattern, the algorithm is probably going to fail, and we want that to be easy to diagnose/debug